<launch>

    <arg name="smallMarkerSize" default="0.25" />    <!-- in m -->
    <arg name="largeMarkerSize" default="0.905" />    <!-- in m -->
    <arg name="robot_name" default="go1" />

    <node pkg="aruco_odometry" type="aruco_odometry_node" name="large_marker_odometry"
        output="screen">
        <remap from="/camera_info" to="/camera/color/camera_info" />
        <remap from="/image" to="/camera/color/image_raw" />
        <param name="image_is_rectified" value="True" />
        <param name="marker_size" value="$(arg largeMarkerSize)" />
        <param name="world_frame" value="world" />
        <param name="robot_frame" value="aruco_camera_link" />
        <rosparam file="$(find aruco_odometry)/cfg/large_marker_poses.yaml" command="load" />
    </node>

    <node pkg="aruco_odometry" type="aruco_odometry_node" name="small_marker_odometry"
        output="screen">
        <remap from="/camera_info" to="/camera/color/camera_info" />
        <remap from="/image" to="/camera/color/image_raw" />
        <param name="image_is_rectified" value="True" />
        <param name="marker_size" value="$(arg smallMarkerSize)" />
        <param name="world_frame" value="world" />
        <param name="robot_frame" value="aruco_camera_link" />
        <rosparam file="$(find aruco_odometry)/cfg/small_marker_poses.yaml" command="load" />
    </node>

    <node pkg="tf" type="static_transform_publisher" name="camera_to_base_link" args="0 0 0 0.5 -0.5 0.5 0.5 /aruco_camera_link /camera_link 100" />
</launch>
